org-brain.el

```diff
@@ -1352,6 +1387,104 @@ After refiling, all headlines will be given an id."
           (org-brain--revert-if-visualizing))
       (org-refile))))
 
+(defun org-brain-refile-to (entry parent)
+  "Refile ENTRY to be a local child of PARENT, returning the new refiled entry.
+
+If ENTRY is linked to PARENT before the refile, this relationship is removed.
+Pins, history, and selected lists are updated
+to account for the change in ENTRY's local parent."
+  (when (member parent (org-brain--local-descendants entry))
+    (error "Cannot refile. New parent %s is a local descendant of %s"
+           (org-brain-title parent) (org-brain-title entry)))
+  (when (org-brain-filep entry)
+    (error "Cannot refile a file entry"))
+  (let ((entry-marker (org-brain-entry-marker entry))
+        (parent-title (org-brain-title parent)))
+    (if (org-brain-filep parent)
+        ;; Parent is a file entry
+        (let ((parent-path (org-brain-entry-path parent)))
+          (with-current-buffer (find-file-noselect parent-path)
+            (goto-char (point-max))
+            (insert "\n* temp headline")
+            (let ((newpoint (point)))
+              (org-with-point-at entry-marker
+                (org-refile nil nil (list parent-title parent-path "" newpoint))))
+            (outline-next-heading)
+            (org-promote-subtree)
+            (outline-previous-heading)
+            (org-cut-subtree)
+            (pop kill-ring)
+            (forward-line -1)
+            (org-brain-remove-line-if-matching "^[[:space:]]*$")))
+      ;; Parent is a headline entry
+      (let ((id (org-brain-entry-identifier parent)))
+        (pcase (org-id-find id)
+          (`(,file-name . ,pos)
+           (org-with-point-at entry-marker
+             (org-refile nil nil (list parent-title file-name "" pos))))
+          (_ (error "Parent headline with ID %s not found" id)))))
+    (let ((new-entry (org-brain-entry-from-id (org-brain-entry-identifier entry))))
+      (cl-flet ((replace-entry (e) (if (equal e entry) new-entry e)))
+        (setq org-brain-pins (mapcar #'replace-entry org-brain-pins))
+        (setq org-brain--vis-history (mapcar #'replace-entry org-brain--vis-history))
+        (setq org-brain-selected (mapcar #'replace-entry org-brain-selected)))
+      (when (member parent
+                    (org-brain--linked-property-entries new-entry "BRAIN_PARENTS"))
+          (org-brain-remove-relationship parent new-entry))
+      (org-save-all-org-buffers)
+      (when (eq major-mode 'org-brain-visualize-mode)
+        (org-brain-visualize new-entry))
```

Comment:
I think the behaviour will be better if doing like this:

```
(when (eq entry 'org-brain--vis-entry)
        (setq org-brain--vis-entry new-entry))
```

The functions calling `org-brain-refile-to` then seems to take care of the "revert if visualizing" part.